File: wear/src/test/java/info/nightscout/androidaps/testing/mockers/RawDataMocker.java
package info.nightscout.androidaps.testing.mockers;

import info.nightscout.androidaps.data.RawDisplayData;
import info.nightscout.androidaps.interaction.utils.WearUtil;
import info.nightscout.shared.SafeParse;
import info.nightscout.shared.weardata.EventData;

@SuppressWarnings("PointlessArithmeticExpression")
public class RawDataMocker {

    private final WearUtilMocker wearUtilMocker;

    public RawDataMocker(WearUtil wearUtil) {
        wearUtilMocker = new WearUtilMocker(wearUtil);
    }

    public RawDisplayData rawSgv(String sgv, int m, String deltaString) {
        RawDisplayData raw = new RawDisplayData();
        double delta = SafeParse.stringToDouble(deltaString);
        String d;

        if (delta <= (-3.5 * 5)) {
            d = "\u21ca";
        } else if (delta <= (-2 * 5)) {
            d = "\u2193";
        } else if (delta <= (-1 * 5)) {
            d = "\u2198";
        } else if (delta <= (1 * 5)) {
            d = "\u2192";
        } else if (delta <= (2 * 5)) {
            d = "\u2197";
        } else if (delta <= (3.5 * 5)) {
            d = "\u2191";
        } else {
            d = "\u21c8";
        }
        raw.setSingleBg(
                new EventData.SingleBg(
                        wearUtilMocker.backInTime(0, 0, m, 0),
                        sgv,
                        "",
                        d,
                        deltaString,
                        "",
                        0,
                        0.0,
                        0.0,
                        0.0,
                        0
                )
        );
        return raw;
    }

    public RawDisplayData rawDelta(int m, String delta) {
        RawDisplayData raw = new RawDisplayData();
        raw.setSingleBg(
                new EventData.SingleBg(
                        wearUtilMocker.backInTime(0, 0, m, 0),
                        "",
                        "",
                        "",
                        delta,
                        "",
                        0,
                        0.0,
                        0.0,
                        0.0,
                        0
                )
        );
        return raw;
    }

    public RawDisplayData rawCobIobBr(String cob, String iob, String br) {
        RawDisplayData raw = new RawDisplayData();
        raw.setStatus(
                new EventData.Status(
                        "",
                        iob,
                        "",
                        true,
                        cob,
                        br,
                        "",
                        "",
                        0L,
                        "",
                        true,
                        0

                )
        );
        return raw;
    }

    public RawDisplayData rawIob(String iob, String iob2) {
        RawDisplayData raw = new RawDisplayData();
        raw.setStatus(
                new EventData.Status(
                        "",
                        iob,
                        iob2,
                        true,
                        "",
                        "",
                        "",
                        "",
                        0L,
                        "",
                        true,
                        0

                )
        );
        return raw;
    }

    public RawDisplayData rawCob(String cob) {
        RawDisplayData raw = new RawDisplayData();
        raw.setStatus(
                new EventData.Status(
                        "",
                        "",
                        "",
                        true,
                        cob,
                        "",
                        "",
                        "",
                        0L,
                        "",
                        true,
                        0

                )
        );
        return raw;
    }

}

File: wear/src/test/java/info/nightscout/androidaps/testing/mockers/RawDataMocker.kt
package info.nightscout.androidaps.testing.mockers

import app.aaps.core.interfaces.rx.weardata.EventData
import app.aaps.core.interfaces.rx.weardata.EventData.SingleBg
import app.aaps.core.interfaces.utils.SafeParse.stringToDouble
import info.nightscout.androidaps.WearTestBase
import info.nightscout.androidaps.data.RawDisplayData

class RawDataMocker {

    fun rawSgv(sgv: String?, m: Int, deltaString: String): RawDisplayData {
        val raw = RawDisplayData()
        val delta = stringToDouble(deltaString)
        val d: String =
            when {
                delta <= -3.5 * 5 -> "\u21ca"
                delta <= -2 * 5   -> "\u2193"
                delta <= -1 * 5   -> "\u2198"
                delta <= 1 * 5    -> "\u2192"
                delta <= 2 * 5    -> "\u2197"
                delta <= 3.5 * 5  -> "\u2191"
                else              -> "\u21c8"
            }

        raw.singleBg = SingleBg(
            WearTestBase.backInTime(0, 0, m, 0),
            sgv!!,
            "",
            d,
            deltaString,
            deltaString,
            "",
            "",
            0,
            0.0,
            0.0,
            0.0,
            0
        )
        return raw
    }

    fun rawDelta(m: Int, delta: String): RawDisplayData {
        val raw = RawDisplayData()
        raw.singleBg = SingleBg(
            WearTestBase.backInTime(0, 0, m, 0),
            "",
            "",
            "",
            delta,
            delta,
            "",
            "",
            0,
            0.0,
            0.0,
            0.0,
            0
        )
        return raw
    }

    fun rawCobIobBr(cob: String, iob: String, br: String): RawDisplayData {
        val raw = RawDisplayData()
        raw.status = EventData.Status(
            "",
            iob,
            "",
            cob,
            br,
            "",
            "",
            0L,
            "",
            0
        )
        return raw
    }

    fun rawIob(iob: String, iob2: String): RawDisplayData {
        val raw = RawDisplayData()
        raw.status = EventData.Status(
            "",
            iob,
            iob2,
            "",
            "",
            "",
            "",
            0L,
            "",
            0
        )
        return raw
    }

    fun rawCob(cob: String?): RawDisplayData {
        val raw = RawDisplayData()
        raw.status = EventData.Status(
            "",
            "",
            "",
            cob!!,
            "",
            "",
            "",
            0L,
            "",
            0
        )
        return raw
    }

}
